src/unleash.ts

```diff
@@ -185,6 +197,16 @@ export class Unleash extends EventEmitter {
     if (!disableAutoStart) {
       process.nextTick(async () => this.start());
     }
+    Unleash.instance = this;
+  }
+
+  static getInstance(config: UnleashConfig) {
+    if(Unleash.instance) {
+      return Unleash.instance;
+    } 
```

Comment:
If you call `getInstance` with a different config, does it make sense to update the config for the instance? Or is that just confusing? It feels a little misleading that you can pass in the config as an argument, but that it does nothing. That said, I'm not sure there is a better way? Could we make it parameterless somehow and have you pass config in other ways? 🤔

Reply:
Good question. I think that you are spot on the `getInstance()` method in the singleton pattern usually does not take arguments (options). I am a bit unsure how/where to communicate the options in a clear and concise way.

Reply:
Yeah, good point. If you remove the need completely and use default values, then I imagine almost everyone will just _not_ configure the instance at all.

Reply:
What about something like:

```
static buildInstance(options?: UnleashConfig){
  if(!options){
     Unleash.singletonConfig = makeDefaultOptions();
  }else {
     Unleash.singletonConfig = options
  }
}

static getInstance() {
  if(!Unleash.singletonConfig){
    throw new Error("You gotta call buildInstance first")
  }
  //setup and/or return instance
}
```

Reply:
you always makes me learn something new @sighphyre, thanks!